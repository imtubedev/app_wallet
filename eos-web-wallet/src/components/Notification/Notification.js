import * as React from "react";
import cx from "classnames";
import css from "./styles.module.scss";

const icons = {
  success: "icon-success",
  info: "icon-info",
  error: "icon-fail_warn",
  warn: "icon-fail_warn"
};

const Notification = ({
  text,
  status = "info",
  minimized = false,
  unsetNotification
}) =>
  !text ? null : (
    <div
      className={cx(css.notification, css[status], { [css.min]: minimized })}
    >
      <div className={cx(icons[status], css.icon)} />
      {text}
      {unsetNotification && (
        <span className={cx("icon-x", css.close)} onClick={unsetNotification} />
      )}
    </div>
  );

export default Notification;
