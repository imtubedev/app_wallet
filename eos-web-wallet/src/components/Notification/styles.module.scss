@import "../../styles/custom_variables.scss";

.notification {
  width: 100%;
  border-radius: 8px;
  color: #fff;
  margin-bottom: 1.5rem;
  padding: 0.875rem 3rem 0.875rem 4.5rem;
  position: relative;
  font-size: 1rem;
  .icon {
    width: 50px;
    height: 100%;
    position: absolute;
    left: 0;
    top: 0;
    border-radius: 6px 0 0 6px;
    text-align: center;
    line-height: 50px;
    font-size: 1.5rem
  }
  .close {
    position: absolute;
    top: 0.875rem;
    right: 1rem;
  }
  &.min {
    margin-bottom: 1rem;
    padding: 0.5rem 1.75rem 0.5rem 2.5rem;
    font-size: 0.75rem;
    .icon {
      width: 30px;
      line-height: 35px;
      font-size: 1rem
    }
    .close {
      top: 0.5rem;
      right: 0.625rem;
    }
  }
}

.error {
  background-color: #574750;
  border: 1px solid $red;

  .icon {
    background-color: $red;
  }
}

.success {
  background-color: #475753;
  border: 1px solid $green;

  .icon {
    background-color: $green;
    color: $grey-80;
  }
}

.warn {
  background-color: #616253;
  border: 1px solid #d8be50;

  .icon {
    background-color: #d8be50;
  }
}

.info {
  background-color: #455b74;
  border: 1px solid $blue;

  .icon {
    background-color: $blue;
    color: white;
  }
}