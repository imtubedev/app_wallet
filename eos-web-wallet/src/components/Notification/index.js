import { connect } from "react-redux";
import Notification from "./Notification";
import { unsetNotification } from "../../redux-modules/notifications/notifications-actions";
import { selectNotificationsState } from "../../redux-modules/notifications/notifications-selectors";

const mapStateToProps = state => selectNotificationsState(state);

const mapDispatchToProps = dispatch => ({
  unsetNotification() {
    dispatch(unsetNotification());
  }
});

export const AppNotifications = connect(mapStateToProps, mapDispatchToProps)(
  Notification
);

export default Notification;
