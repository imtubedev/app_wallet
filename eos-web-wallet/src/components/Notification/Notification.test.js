import * as React from "react";
import renderer from 'react-test-renderer';
import Notification from './';

describe('<Notification />', () => {
  it('renders without crashing', () => {
    const notification = {
      text: 'Connection error',
      status: 'error',
      unsetNotification: () => {},
    };
    const tree = renderer.create(
      <Notification {...notification} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
})


