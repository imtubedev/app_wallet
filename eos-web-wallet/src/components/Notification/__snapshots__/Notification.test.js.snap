// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Notification /> renders without crashing 1`] = `
<div
  className=""
>
  <div
    className="icon-fail_warn"
  />
  Connection error
  <span
    className="icon-x"
    onClick={[Function]}
  />
</div>
`;
